mailpoet/lib/Newsletter/Sending/SendingQueuesRepository.php

```diff
@@ -48,6 +48,17 @@ public function findOneByNewsletterAndTaskStatus(NewsletterEntity $newsletter, s
       ->getOneOrNullResult();
   }
 
+  public function getTaskIdsByNewsletterId(int $newsletterId): array {
+    $results = $this->entityManager->createQueryBuilder()
+      ->select('IDENTITY(s.task) as task_id')
```

Comment:
I wanted to select only taks_ids but you can't do it in DQL directly because it works with the associated entity as a whole (SendingQueueEntity has $task property but not $task_id). IDENTITY is a [DQL function](https://www.doctrine-project.org/projects/doctrine-orm/en/latest/reference/dql-doctrine-query-language.html#dql-functions) that is interpreted by Doctrine and can be used for retrieving association column values.

Reply:
Thanks for the clarification. I wasn't aware that DQL functions were a thing.